frame/staking/src/lib.rs

```diff
@@ -3225,12 +2395,14 @@ impl<T: Config> Module<T> {
 	/// Enact and process the election using the `ElectionProvider` type.
 	///
 	/// This will also process the election, as noted in [`process_election`].
-	fn enact_election(_current_era: EraIndex) -> Option<Vec<T::AccountId>> {
-		let _outcome = T::ElectionProvider::elect().map(|_| ());
-		log!(debug, "Experimental election provider outputted {:?}", _outcome);
-		// TWO_PHASE_NOTE: This code path shall not return anything for now. Later on, redirect the
-		// results to `process_election`.
-		None
+	fn enact_election(current_era: EraIndex) -> Option<Vec<T::AccountId>> {
+		T::ElectionProvider::elect()
+			.map_err(|e| log!(warn, "election provider failed due to {:?}", e))
+			.and_then(|(r, w)| {
+				<frame_system::Pallet<T>>::register_extra_weight_unchecked(w, frame_support::weights::DispatchClass::Mandatory);
```

Comment:
This should be a follow up issue: doing this here would lead to adding weight to the interface of SessionManager, and SessionInterface, which I don't want to do in this critical PR.